Add main page

Ramazanmak opened the pull request.

Ramazanmak committed:
feat: make hero-section and features-section
- make header with mobile menu;
- make adaptive hero-section;
- make section with features where intuitive syntax changed to consistent one

mr150 commented on src/_includes/components/card.ejs:
```diff
@@ -0,0 +1,57 @@
+	let cardBg = typeof bg !== 'undefined' ? bg : 'Bgc-$featuresBgc'
+	let cardLogoBg = typeof logoBg !== 'undefined' ? logoBg : 'Bgc-$featuresBgc'
+	let cardTitleC = typeof headC !== 'undefined' ? headC : 'Example title'
+	let cardTextC = typeof textC !== 'undefined' ? textC : 'Example text'
```
Зачем это? Мы же в конфиге eleventy делали `localsName: 'it'`

mr150 commented on src/_includes/components/card.ejs:
```diff
@@ -0,0 +1,57 @@
+@:pfcs-l_-ModLeftBgc-rgba(243,232,255,1) @:pfcs-l_-ModRightBgc-rgba(233,212,255,0.5)
+@:pfcs-l_-ModLeftBgc-rgba(233,212,255,1)_h @:pfcs-l_-ModRightBgc-rgba(218,178,255,0.5)_h
+@:pfcs-l_-ModHeadC-rgb(89,22,139) @:pfcs-l_-ModTextC-rgb(130,0,219)
+-ModLeftLogoBgc-rgb(173,70,255) -ModRightLogoBgc-rgb(133,0,226)
```
Не надо делать цвета в rgb. Все цвета: либо в css переменных (приоритет), либо ключевыми словами в конфиге. Такое только как костыль в единичном случае

mr150 commented on src/_includes/components/link-button.ejs:
```diff
@@ -0,0 +1,46 @@
+	let buttonVariant = typeof variant !== 'undefined' ? variant : 'light'
+%>
+
+<% if (buttonVariant === "dark") {%>
```
Зачем здесь копипаст? Разве нельзя просто значения переменных переключить?

mr150 commented on src/_includes/header/burger.ejs:
```diff
@@ -0,0 +1,11 @@
+<label for="menu-toggle"
+  class=" D P1.5u Bd-n C-$brandC Bdrd1u Pt0 H90p Apcr1 Bgc-$headerBgc Ps-r Bgc-$headerBgc
+   ^burger:c:+_Bgc-$brandC Tsd-$transD ^burger:c:+_Bdc-$brandC Fns1.5e ^burger:c:+_C-$headerBgc ^burger:c:+_Bdc-n Zi4">
+  ☰
```
Это псевдоэлементом делается

mr150 commented on src/_includes/header/header.ejs:
```diff
@@ -0,0 +1,55 @@
+	<nav id="desktop-menu"
+		class="Jc-c Ai-c Flg0.5 C-$headerFontC W50p D-n Fns-$headerFns
+		md_D-f md_Flg0.5">
+  	<a href="/#features" class="D Txd-n M-0;2.5u P1u;3u C-$headerFontC C-$brandC_h
```
`/` здесь лишний

mr150 commented on style.min.css:
`min` файлы не коммитим

mr150 commented on src/layouts/base.ejs:
```diff
@@ -1,20 +1,32 @@
+  <title>MLUT</title>
 </head>
-<body class="Mnh100vh D-f Fld-c Jc-sb Bgc#f5f5f5 Fnf-sans-serif">
+<body class="D-f Fld-c Jc-sb Fnf-sans-serif Bxs-bb Ovx-h Bgc-$bodyBgc">
```
`Bxs-bb`, уже есть по дефолту, а `Ovx-h` - костыль, который скрывает баги верстки. Если баги есть, то надо их фиксить, а не прятать

mr150 reviewed (changes requested):
Нужны фиксы

Ramazanmak committed:
feat: fix everything considering PR remarks and make examples section

Ramazanmak committed:
feat: make CSS Art gallery

Ramazanmak committed:
feat:make footer

mr150 commented on src/_includes/components/arts/ghost.ejs:
```diff
@@ -0,0 +1,51 @@
+<div id="canva"
+  class="Apcr1 W100p Ps M-a D-f Ai-c Jc-c Ctnn-canva Ctnt-s ">
+
+  <div id="circle"
```
Для чего тут и во многих других местах id?

mr150 commented on src/_includes/components/arts/ghost.ejs:
```diff
@@ -0,0 +1,51 @@
+      class="Ps-r Bd0.5vmin;s;black W100p H100p Bdtlr50p Bdtrr50p Bdb0 Bxs-bb Bgc#FFDCDF">
+
+      <div id="heart"
+        class="Ps-r W10p H10p Bgc#FF3737 Tf-translate(700%,200%);rotate(25deg);scale(0.8)
```
transform через отдельные кастомные утилиты делается

mr150 commented on src/_includes/components/arts/ghost.ejs:
```diff
@@ -0,0 +1,51 @@
+
+      <div id="heart"
+        class="Ps-r W10p H10p Bgc#FF3737 Tf-translate(700%,200%);rotate(25deg);scale(0.8)
+        Ps-a_b B0_b L-50p_b W100p_b H100p_b Ct_b Bdrd50p_b Bgc#FF3737_b
```
Не надо делать hex цвета капсом

mr150 commented on src/index.ejs:
```diff
@@ -1,10 +1,322 @@
+			<img class="D" alt="mlut logo" src="/assets/img/logo.png">
+		</picture>
+
+		<h1 class="Txa-c C-$accent900 Fns8u Lnh1.3 M0 P5u;1u
```
`h1` не должен быть внутри секции

Ramazanmak committed:
fix: fixed bugs
> - replaced id's with comments,
> - instead of mutating it included new local variables,
> - refactored link-button component,
> - and made some stylistical changes.

mr150 commented on src/_includes/components/arts/ghost.ejs:
```diff
@@ -0,0 +1,41 @@
+<%
+	const ghostStyles = {
```
Словари с алиасами лучше заканчивать на `css` для унификации. Типа `ghostCss`

mr150 commented on src/_includes/components/arts/ghost.ejs:
```diff
@@ -0,0 +1,41 @@
+	const ghostStyles = {
+		eye: "Ps-a W25p H25p Bgc#000 Bdrd50p T45p Ct_b Ps-a_b Bgc#fff_b W40p_b H40p_b Bdrd50p_b L40p_b T10p_b",
+		hand: "Ps-a W20p H20p Bd0.5vmin;s;black Bdblr100 Bdbrr100 Bdt0 T78p",
+		leg: "W25p H100p Bd0.5vmin;s;black Bdblr100 Bdbrr100 Bdt0 Bgc#ffdcdf",
```
Для `border-radius` тоже есть кастомные утилиты на сокращенные свойства для top и bottom: `-Bdbrd`

mr150 commented on src/_includes/components/arts/ghost.ejs:
```diff
@@ -0,0 +1,41 @@
+    <div class="Ps-r Bd0.5vmin;s;black W100p H100p Bdtlr50p Bdtrr50p Bdb0 Bxs-bb Bgc#ffdcdf">
+
+			<%# heart %>
+      <div class="Ps-r W10p H10p Bgc#ff3737 -Trx700p -Try200p -Rt25d Sx0.8 Sy0.8 Tf
```
Тут цвет повторяется. Его надо куда-то вынести

mr150 replied:
Для `scale` то же есть сокращенное свойство `-S`

mr150 commented on src/_includes/components/arts/penrose-triangle.ejs:
```diff
@@ -0,0 +1,52 @@
+	<%# left-black %>
+
+  <div class="Apcr1 Ps-a @c:canva:apcr>=1_H50p @c:canva:apcr<=1_W50p
+    -Gdl-60deg,tp;19.5p,#444444;19.5p,#444444;32p,tp;32p Bgp0p;34.5p Bgr-nr
```
Зачем `#444444`, если можно `#444`

mr150 commented on src/_includes/components/arts/penrose-triangle.ejs:
```diff
@@ -0,0 +1,52 @@
+
+	<%# right-white2 %>
+
+  <div class="Apcr1 Ps-a @c:canva:apcr>=1_H50p  @c:canva:apcr<=1_W50p
```
Этот набор утилит кучу раз повторяется - надо вынести

mr150 commented on src/_includes/footer.ejs:
```diff
@@ -1,5 +1,72 @@
+<footer class="P8u;0 W100p Bgc-$core850 C#d1d5db D-g Gtc1fr;1fr Gtr0.2fr;1fr;0.2fr Gap2u  md_Gtc0.5fr;1fr;1fr;1fr;0.5fr md_Gtr1fr;0.2fr md_Gap5u" >
+
+  <section class="W90p M-a Gcs1 Gce3 Grs1 Gre2 md_Gcs2 md_Gce3 ">
+		<picture class="D W40p Mxw30u P2u;1u xl_Mxw56u
```
`picture` здесь все еще бессмысленный

mr150 commented on src/_includes/footer.ejs:
```diff
@@ -1,5 +1,72 @@
+		</h3>
+	</section>
+
+	<section class="<%= styles.linkSect %>  Gcs1 Gce2 Grs2 Gre3 md_Grs1 md_Gre2 md_Gcs3 md_Gce4">
```
Сложновато раскладка выглядит. На флексах тут не проще было бы?

mr150 commented on src/index.ejs:
```diff
@@ -1,10 +1,346 @@
+	const css = {
+		codeWord: "Bgc-$core800 P0.3u;1u"
+	}
+	const heroCSS = {
```
Нет смысла делать 2 словаря для алиасов

Ramazanmak committed:
fix: fix bugs according to the comments:
> - change styles in the footer;
> - include nav links using lists;
> - make header and universal;
>

mr150 commented on src/_includes/components/arts/ghost.ejs:
```diff
@@ -0,0 +1,42 @@
+    <div class="Ps-r Bd0.5vmin;s;black W100p H100p Bdtlr50p Bdtrr50p Bdb0 Bxs-bb Bgc-$bodyBgc">
+
+			<%# heart %>
+      <div class="Ps-r W10p H10p Bgc-$heartBgc -Trx700p -Try200p -Rt25d Sx0.8 Sy0.8 Tf
```
Не везде используются утилиты для сокращенных свойств: `-S` и `-Sz`

mr150 commented on src/_includes/footer.ejs:
```diff
@@ -1,5 +1,59 @@
+		linkSect:"P5u W49p md_W31p D-f Fld-r Flw-w Jc-fs md_Jc-c Ai-fs md_Ai-fs md_P0;3u md_W30p",
+		ul:"P0 Lsst-n W70p",
+		li:"P1u;0",
+		link:"Txd-n C#d1d5db C-$brand_h Tsd-$shortTs",
```
Снова повторяющийся цвет без переменной

mr150 commented on src/_includes/header/header.ejs:
```diff
@@ -0,0 +1,80 @@
+
+		<%# External links %>
+
+			<h2 class="P10u;5u;3u Txa-r M0 C-$brand md_D-n"> Find us</h2>
```
Не должно быть 2 h2 внутри одного `nav`. Второй список - скорее всего что-то отдельное

Ramazanmak replied:
Вот эти два h2: "menu" и "find us"
<img width="403" height="592" alt="image" src="https://github.com/user-attachments/assets/0b1ae066-fb42-4d23-8a01-23865d75f8c1" />

mr150 commented on src/index.ejs:
```diff
@@ -1,10 +1,346 @@
+<%# Examples Section %>
+
+<%
+	const examplesCSS = {
```
Много отдельных словарей с алиасами. Ничего из их содержимого не пересекается?

Ramazanmak replied:
Да вроде не очень много - всего три словаря на весь index.ejs (один только большой - для examples секции). А на все файлы со всеми компонентами - штук 8-9. Пересекаются они мало, если честно, а там, где они пересекаются - делают это немного, поэтому я не особо вижу смысл заводить ещё и глобальный словарь.

Ramazanmak committed:
fix: fix some minor bugs

mr150 commented on src/_includes/components/arts/penrose-triangle.ejs:
```diff
@@ -0,0 +1,57 @@
+
+	<%# right-white2 %>
+
+  <div class="Apcr1 Ps-a <%= css.div %>
```
Почему вынесена только часть, если повторяются все утилиты тут? И зачем для одного алиаса словарь, если достаточно обычной переменной

mr150 commented on src/_includes/components/arts/penrose-triangle.ejs:
```diff
@@ -0,0 +1,57 @@
+<%
+	const css = {
+		div: "@c:canva:apcr>=1_H50p  @c:canva:apcr<=1_W50p  Bgr-nr"
```
И алиасам надо давать осмысленные названия, а не `div`

mr150 commented on src/_includes/header/burger.ejs:
```diff
@@ -0,0 +1,12 @@
+</label>
+
+<label id="overlay" for="menu-toggle"
+  class="Ps-a md_D-n W100vw Tsd-$transD H100vh T0 R0 D Bgc-n ^burger:!c:~_Pne-n ^burger:c:~_Bgc#000*60p">
```
Здесь тоже особо нет смысла в именованном контексте

Ramazanmak replied:
У меня почему-то не работает `:!c:~_Pne-n`, хотя `:c:~_Bgc#000*60p` - работает отлично. Может, потому что мы, убирая контекст ^burger, переносим 'слушатель события' c чекбокса на сам label, а у него при неактивном чекбоксе отключены pointer-events, и поэтому при нажатии у него pointer-events и не включаются?

Но я по-другому поступил: вместо `D ^burger:!c:~_Pne-n`, написал `D-n :c:~_D, и вроде всё работает.

Ramazanmak committed:
fix: fix some bugs:
- change padding to margins where needed;
delete context where they're unnecessary.

Ramazanmak committed:
fix: change the layout
- simplify hero-section;
- reorganize examples sections;
- enrich code examples;
- add new art.

Ramazanmak committed:
fix: make some UI changes:
- change hero-page;
- make header more convenient;
- improve the cross-browser compatibility of the arts.

Ramazanmak committed:
fix: make minor fixes:
- change sizes of the arts;
- change size of the logo in hero-section;
- change the caption on the button in the header.

Ramazanmak committed:
fix: make stylistical fixes:
- change the style of the code samples;
- remove extra code samples;
- add new subsections.

Ramazanmak committed:
refactor: make some minor stylistical changes

mr150 commented on src/_includes/components/card.ejs:
```diff
@@ -0,0 +1,25 @@
+
+<article class=" <%= it.bg %> <%= it.bgH %> P7u Bdrd2u Mxw110u Tsd0.5s
```
`article` здесь не в тему. Оно в списке фич находится. Должно быть опциональное имя тега с `li` по умолчанию

Ramazanmak replied:
То есть мне можно просто обернуть в тег <li>, изменив контейнер на <ul>? Или в компоненте сделать дополнительный входной параметр, который по умолчанию даёт <li>, а так может и любой вводимый тег вставить?

mr150 commented on src/_includes/components/code-samples/regcss1.ejs:
```diff
@@ -0,0 +1,30 @@
+&lt;/div&gt;
+
+&lt;style&gt;
+  .parent{
```
```suggestion
  .parent {
```

mr150 commented on src/index.ejs:
```diff
@@ -1,10 +1,437 @@
+
+		<%# Naming conventions part%>
+
+		<article class="<%= examplesCSS.article%>">
```
`article` здесь не в тему

mr150 commented on src/layouts/base.ejs:
```diff
@@ -1,20 +1,32 @@
+	<link rel="stylesheet" href="/assets/style/style-hljs.css">
+
+	<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.11.1/styles/github-dark.min.css">
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.11.1/highlight.min.js"></script>
```
Скрипты должны загружаться в конце body

mr150 commented on src/_includes/footer.ejs:
```diff
@@ -1,5 +1,60 @@
+
+		<section class="W100p D-f P0;5u Ai-c Jc-fs md_Jc-c ">
+			<p class="C-$gray100 Txa-l M0 ">
+				© 2025 mlut CSS Framework. All rights reserved.
```
```suggestion
				© 2025 mlut contributors. All rights reserved.
```

mr150 commented on src/layouts/base.ejs:
```diff
@@ -1,20 +1,32 @@
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.11.1/highlight.min.js"></script>
+
+	<%# and it's easy to individually load additional languages %>
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.11.1/languages/go.min.js"></script>
```
Это скрипт для golang?)

Ramazanmak committed:
refactor:
- improve page's semantics;
- change content in some places;
- reorganize base.ejs structure.

mr150 commented on src/_includes/footer.ejs:
```diff
@@ -51,9 +51,16 @@
 		</section>
 
-		<section class="W100p D-f P0;5u Ai-c Jc-fs md_Jc-c ">
+		<section class="W100p D-f P0;5u Fld-c Ai-c Jc-fs md_Jc-c ">
```
В `section` обязательно должен быть загаловок (визуально скрытый тоже подойдет). Если осмысленного заголовка нет - это не `section`

mr150 commented on src/layouts/base.ejs:
```diff
@@ -1,20 +1,29 @@
+	<link rel="stylesheet" href="/assets/style/style-hljs.css">
+
+	<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.11.1/styles/github-dark.min.css">
+  <title>MLUT</title>
```
`mlut` - всегда и везде пишется только так, без заглавных букв

Ramazanmak committed:
refactor: make some changes according to the comments
